install_files/ansible-base/roles/tails-config/files/securedrop_init.py

```diff
@@ -148,3 +150,71 @@
 if b'Update needed' in output or os.path.exists(flag_location):
     # Start the SecureDrop updater GUI.
     subprocess.Popen(['python3', path_gui_updater], env=env)
+
+# Check for Tails < 4.19 and apply a fix to the auto-updater.
+# See https://tails.boum.org/news/version_4.18/
+# (Suggested removal: 2022/01)
+tails_min_version = [4, 19]
+needs_update = False
+
+try:
+    cmd = 'cat /etc/os-release | grep VERSION | cut -f2 -d\\"'
+
+    # Using shell=True because contents of /etc/os-release are trusted
+    tails_current_version = subprocess.check_output(cmd,
+                                                    shell=True,
+                                                    universal_newlines=True,
+                                                    env=env).strip().split(".")  # nosec
+
+    try:
+        needs_update = (len(tails_current_version) >= len(tails_min_version) and
+                        (int(tails_current_version[0]) < tails_min_version[0]
+                        or int(tails_current_version[1]) < tails_min_version[1]))
+
+    except (TypeError, ValueError):
+        sys.exit('Error checking Tails version. Please visit tails.boum.org ' +
+                 'to ensure your version of Tails is up to date.')
+
+    if needs_update:
+        cert_name = 'isrg-root-x1-cross-signed.pem'
+        pem_file = tempfile.NamedTemporaryFile(delete=True)
```

Comment:
Just one note on this draft to inform the test plan:  The `pem_file.name` to look for under `/tmp/` will not be the value of `cert_name`.  That doesn't matter for production use, but to ease testing you could do something like `tempfile.NamedTemporaryFile(prefix=cert_name, ...)`.

Reply:
Thanks - changing the test plan instead, since the auto-updater appearing is really the important part :)